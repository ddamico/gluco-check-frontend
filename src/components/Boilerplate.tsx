import React from "react";
import { Trans, useTranslation } from "react-i18next";
import { Link, makeStyles, Typography } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  container: {
    "& p": {
      fontSize: ".5rem",
    },
  },
}));

function Boilerplate() {
  const { t } = useTranslation();
  const classes = useStyles();
  return (
    <div className={classes.container}>
      <Typography>
        <Trans i18nKey="boilerplate.terms">
          By continuing, you are indicating that you accept our{" "}
          <Link href={t("urls.termsAndConditions")}>Terms of Service</Link> and{" "}
          <Link href={t("urls.privacy")}>Privacy Policy</Link>.
        </Trans>
      </Typography>
      <Typography>{t("boilerplate.google")}</Typography>
      <Typography>
        <Trans i18nKey="boilerplate.nightscout">
          Not affiliated with the{" "}
          <Link href={t("urls.nightscoutProject")}>Nightscout Project</Link>
        </Trans>
      </Typography>
    </div>
  );
}

export default Boilerplate;
