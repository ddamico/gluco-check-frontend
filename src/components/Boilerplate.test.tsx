import React from "react";
import { cleanup, render } from "@testing-library/react";
import { axe } from "jest-axe";
import Boilerplate from "./Boilerplate";

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
    };
  },
  Trans: function (props: any) {
    return <span>{props.i18nKey}</span>;
  },
}));

afterEach(cleanup);

describe("Landing page", () => {
  it("renders the component", () => {
    const { container } = render(<Boilerplate />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    const { container } = render(<Boilerplate />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
