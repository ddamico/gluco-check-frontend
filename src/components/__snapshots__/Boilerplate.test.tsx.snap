// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page renders the component 1`] = `
<div
  class="makeStyles-container-1"
>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span>
      boilerplate.terms
    </span>
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    boilerplate.google
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span>
      boilerplate.nightscout
    </span>
  </p>
</div>
`;
